examples/dmd/wave_equation.cpp

```diff
@@ -0,0 +1,685 @@
+
+      if (last_step || (ti % windowNumSamples) == 0)
+      {
+         cout << "step " << ti << ", t = " << t << endl;
+
+         if(rdim != -1){
+            
+            if(myid == 0){
+
+               std::cout << "Creating DMD with rdim: " << rdim << std::endl;
+            }
+            
+            dmd_u[curr_window]->train(rdim);
+         }
+         
+         else if(ef != -1){
+            if(myid == 0){
+               std::cout << "Creating DMD with energy fraction: " << ef << std::endl;
+            }
+            dmd_u[curr_window]->train(ef);
+         }
+
+         if(!last_step){
+             curr_window++;
+             dmd_u.push_back(new CAROM::DMD(u_gf.GetTrueVector().Size(), dt));
+             dmd_u[curr_window]->takeSample(u_gf.GetTrueVector(), t);
+         }
+
+      }
+      ts.push_back(t);
+      dmd_training_timer.Stop();
+       
+     if (last_step || (ti % vis_steps) == 0)
+        {
+            
+            u_gf.SetFromTrueDofs(u);
+            dudt_gf.SetFromTrueDofs(dudt);
+
+            if (visualization)
+            {
+               sout << "solution\n" << *mesh << u_gf << flush;
+            }
+
+            if (visit)
+            {
+               visit_dc.SetCycle(ti);
+               visit_dc.SetTime(t);
+               visit_dc.Save();
+            }
+        }
+       oper.SetParameters(u);
+
+   }
+
+   // 9. Save the final solution. This output can be viewed later using GLVis:
+   //    "glvis -m wave_equation.mesh -g wave_equation-final.gf".
+   {
+      cout << "Step 9" << endl;
+      ofstream osol("wave_equation-final.gf");
+      osol.precision(precision);
+      u_gf.Save(osol);
+      dudt_gf.Save(osol);
+   }
+
+   // 10. Calculate the DMD modes.
+    if (myid == 0 && rdim != -1 && ef != -1)
+    {
+        std::cout << "Both rdim and ef are set. ef will be ignored." << std::endl;
+    }
+
+   //Vector true_solution_u(u_gf.Size());
+
+   Vector true_solution_u(u_gf.GetTrueVector().Size());
+   
+
+   //true_solution_u = u.GetData();
+
+   true_solution_u = u_gf.GetTrueVector();
+
+
+   dmd_prediction_timer.Start();
+
+   // 11. Predict the state at t_final using DMD.
+   cout << "Step 11" << endl;
+   if (myid == 0)
+   {
+      std::cout << "Predicting temperature using DMD" << std::endl;
+   }
+
+   curr_window = 0;
+   
+   CAROM::Vector* result_u = dmd_u[curr_window]->predict(ts[0]);
+   Vector initial_dmd_solution_u(result_u->getData(), result_u->dim());
+   u_gf.SetFromTrueDofs(initial_dmd_solution_u);
+   //dudt_gf.SetFromTrueDofs(dudt);
+
+   VisItDataCollection dmd_visit_dc("DMD_Wave_Equation", mesh);
+   dmd_visit_dc.RegisterField("solution", &u_gf);
+   //dmd_visit_dc.RegisterField("rate", &dudt_gf);
+   if (visit)
+   {
+      dmd_visit_dc.SetCycle(0);
+      dmd_visit_dc.SetTime(0.0);
+      dmd_visit_dc.Save();
+   }
+
+   delete result_u;
+
+   for (int i = 1; i < ts.size(); i++)
+   {
+      result_u = dmd_u[curr_window]->predict(ts[i]);
```

Comment:
Please introduce if(visit) statement here so that dmd "predict" is called only when visit flag is true.

Reply:
Done.